plugins/serializers/template/template_test.go

```diff
@@ -0,0 +1,131 @@
+package template
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+)
+
+var basicMetric = metric.New(
+	"cpu",
+	map[string]string{
+		"host": "localhost",
+		"cpu":  "CPU0",
+	},
+	map[string]interface{}{
+		"x": 42.0,
+		"y": 43.0,
+	},
+	time.Unix(100, 0),
+)
+
+var tests = []struct {
+	name      string
+	input     telegraf.Metric
+	template  string
+	output    []byte
+	errReason string
+}{
+	{
+		name:     "name",
+		input:    basicMetric,
+		template: "{{ .Name }}",
+		output:   []byte("cpu"),
+	},
+	{
+		name:     "time",
+		input:    basicMetric,
+		template: "{{ .Time.Unix }}",
+		output:   []byte("100"),
+	},
+	{
+		name:     "specific field",
+		input:    basicMetric,
+		template: `{{ .Field "x" }}`,
+		output:   []byte("42"),
+	},
+	{
+		name:     "specific tag",
+		input:    basicMetric,
+		template: `{{ .Tag "cpu" }}`,
+		output:   []byte("CPU0"),
+	},
+	{
+		name:     "all fields",
+		input:    basicMetric,
+		template: `{{ range $k, $v := .Fields }}{{$k}}={{$v}},{{end}}`,
+		output:   []byte("x=42,y=43,"),
+	},
+	{
+		name:     "all tags",
+		input:    basicMetric,
+		template: `{{ range $k, $v := .Tags }}{{$k}}={{$v}},{{end}}`,
+		output:   []byte("cpu=CPU0,host=localhost,"),
+	},
+	{
+		name:     "string",
+		input:    basicMetric,
+		template: "{{ .String }}",
+		output:   []byte("cpu map[cpu:CPU0 host:localhost] map[x:42 y:43] 100000000000"),
+	},
+	{
+		name: "complex",
+		input: metric.New(
+			"cpu",
+			map[string]string{
+				"tag1": "tag",
+			},
+			map[string]interface{}{
+				"value": 42.0,
+			},
+			time.Unix(0, 0),
+		),
+		template: `{{ .Name }} {{ range $k, $v := .Fields}}{{$k}}={{$v}}{{end}} {{ .Tag "tag1" }} {{.Time.UnixNano}} literal`,
+		output:   []byte("cpu value=42 tag 0 literal"),
+	},
+}
```

Comment:
Please define this in the corresponding test to make it self-contained and to avoid potential side-effects...